back/engines/free/email_campaigns/app/mailers/email_campaigns/application_mailer.rb

```diff
@@ -11,6 +11,7 @@ class ApplicationMailer < ApplicationMailer
     end
 
     def campaign_mail
+      locale = params[:locale] || locale
```

Comment:
This seems to override the recipient locale, meaning all mails are sent in the same locale and are not sensitive to the recipient's locale.

I've added a test to this PR to illustrate this and for you use in checking the functionality of your changes, though I just tested one campaign mail, so you might want to make a test that's more general).

You can also reproduce the issue like this:
1. Run app locally
2. Set locale to language A and sign up user_1
3. Set locale to language B and sign up user_2
4. Submit an Idea to 'An Idea? ...' project as user_1
5. Submit an Idea to 'An Idea? ...' project as user_2
6. Check the mails sent to each user (Subject: 'Your idea...', or equivalent in different language, in mailcatcher (localhost:1080)
=> both mails are in same language, but should be in different languages (those of the the locales of each user)

Reply:
thanks, i tested very poorly it seems. i've added tests for the emails that are not campaigns now.